Prox/Prox/AppDelegate.swift

```diff
@@ -23,9 +26,28 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 
         // display
         window?.makeKeyAndVisible()
+
+        setupFirebase()
+
         return true
     }
 
+    private func setupFirebase() {
+        FIRApp.configure()
+
+        if let user = FIRAuth.auth()?.currentUser {
+            authorizedUser = user
+        } else {
+            FIRAuth.auth()?.signInAnonymously { (user, error) in
```

Comment:
yes you are. It's an interesting feature of swift, if the final argument of a function is a closure, you can define that closure at the end of the function, not inside it, and you don't have to specify it's var name. This _is_  [`signInAnonymouslyWithCompletion`](https://firebase.google.com/docs/reference/ios/firebaseauth/interface_f_i_r_auth.html#a0ffa43acd046e5ee42967106f73556f6)